src/actions/project-text-search.js

```diff
@@ -48,12 +48,12 @@ export function searchSources(query: string) {
     await dispatch(clearSearchResults());
     await dispatch(addSearchQuery(query));
     dispatch(updateSearchStatus(statusType.fetching));
-    await dispatch(loadAllSources());
```

Comment:
Like the shift in approach. Instead of loading everything up front we load them one by one.